feature/experimental/bgp/otg_tests/base_bgp_session_parameters/base_bgp_session_parameters_test.go

```diff
@@ -0,0 +1,526 @@
+	dev := topo.Devices().Add().SetName(ateAttrs.Name)
+	eth := dev.Ethernets().Add().SetName(ateAttrs.Name + ".Eth")
+	eth.Connection().SetPortName(port1.ID())
+	eth.SetMac(ateAttrs.MAC)
+
+	ip := eth.Ipv4Addresses().Add().SetName(dev.Name() + ".IPv4")
+	ip.SetAddress(ateAttrs.IPv4).SetGateway(dutAttrs.IPv4).SetPrefix(int32(ateAttrs.IPv4Len))
+
+	bgp := dev.Bgp().SetRouterId(ateAttrs.IPv4)
+	peerBGP := bgp.Ipv4Interfaces().Add().SetIpv4Name(ip.Name()).Peers().Add()
+	peerBGP.SetName(ateAttrs.Name + ".BGP4.peer")
+	peerBGP.Advanced().SetHoldTimeInterval(ateHoldTime)
+	if auth == md5Auth {
+		peerBGP.Advanced().SetMd5Key(authPassword)
+	}
+	peerBGP.SetPeerAddress(ip.Gateway()).SetAsNumber(int32(ateParams.localAS))
+	if connectionType == connInternal {
+		peerBGP.SetAsType(gosnappi.BgpV4PeerAsType.IBGP)
+	} else {
+		peerBGP.SetAsType(gosnappi.BgpV4PeerAsType.EBGP)
+	}
+
+	return topo
+
+}
+
+// createCeaseAction creates the BGP cease notification action in gosnappi
+func createCeaseAction(t *testing.T) gosnappi.ControlAction {
+	t.Helper()
+	ceaseAction := gosnappi.NewControlAction().SetChoice(gosnappi.ControlActionChoice.PROTOCOL)
+	ceaseAction.Protocol().SetChoice(gosnappi.ActionProtocolChoice.BGP).Bgp().
+		SetChoice(gosnappi.ActionProtocolBgpChoice.NOTIFICATION).
+		Notification().SetNames([]string{}).
+		SetChoice(gosnappi.ActionProtocolBgpNotificationChoice.CUSTOM).
+		Custom().
+		SetCode(6).
+		SetSubcode(6)
+	return ceaseAction
+}
+
+// TestEstablishAndDisconnect Establishes BGP session between DUT and ATE and Verifies
+// abnormal termination of session using notification message:
+func TestEstablishAndDisconnect(t *testing.T) {
+	// DUT configurations.
+	t.Log("Start DUT config load:")
+	dut := ondatra.DUT(t, "dut")
+
+	// Configure interface on the DUT
+	t.Log("Start DUT interface Config")
+	configureDUT(t, dut)
+
+	// Configure Network instance type on DUT
+	t.Log("Configure Network Instance")
+	dutConfNIPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
+	gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+
+	t.Log("Configure BGP")
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	nbrPath := statePath.Neighbor(ateAttrs.IPv4)
+
+	bgpClearConfig(t, dut)
+	dutConf := bgpCreateNbr(&bgpTestParams{localAS: dutAS, peerAS: ateAS}, dut)
+	gnmi.Replace(t, dut, dutConfPath.Config(), dutConf)
+	fptest.LogQuery(t, "DUT BGP Config", dutConfPath.Config(), gnmi.GetConfig(t, dut, dutConfPath.Config()))
+
+	// ATE Configuration.
+	t.Log("Configure port and BGP configs on ATE")
+	ate := ondatra.ATE(t, "ate")
+	topo := configureATE(t, &bgpTestParams{localAS: ateAS, peerIP: dutAttrs.IPv4}, connExternal, noAuth)
```

Comment:
The ATE version of the test does still use MD5 auth for the TestEstablishAndDisconnect test case (which matches the latest README for that test.) Can we update this test to do the same?

Reply:
@octpetre : Can we get this update?

Reply:
We removed the MD5 part from TestEstablishAndDisconnect because this feature is not yet supported in OTG. The README for the otg version was also updated to not use MD5. This is tested separately in the TestPassword function which as of now is skipped in the otg version.